Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2015,10 +1995,79 @@ extension BrowserViewController: IntroViewControllerDelegate {
         return false
     }
     
-    func onboardingUserResearchHelper() {
+    private func onboardingUserResearchHelper(_ forcedType: OnboardingScreenType? = nil) {
         print("lp initial value \(String(describing: onboardingUserResearch?.lpVariable?.boolValue()))")
+        if forcedType != nil {
+            showProperIntroVC(forcedType)
+            return
+        }
+        let screenType = onboardingUserResearch?.onboardingScreenType
+        if screenType == nil && !DeviceInfo.hasConnectivity() {
+            self.onboardingUserResearch?.updateValue(value: true)
+            showProperIntroVC()
+            return
+        }
         onboardingUserResearch?.updatedLPVariables = {(lpVariable) -> () in
-            print("lpVariable \(String(describing: lpVariable?.boolValue()))")
+            if screenType == nil {
+                print("lp Variable from server \(String(describing: lpVariable?.boolValue()))")
+                self.onboardingUserResearch?.updateTelemetry()
+                self.onboardingUserResearch?.updateValue(value: lpVariable?.boolValue() ?? true)
+                self.showProperIntroVC()
+            }
+        }
+    }
+    
+    private func showProperIntroVC(_ forcedType: OnboardingScreenType? = nil) {
```

Comment:
**Refactoring Note:** Will likely be moved to a new ViewModel for when we combine introViewController + introViewControllerV2